src/Services/EncounterService.php

```diff
@@ -307,7 +307,7 @@ public function insertEncounter($puuid, $data)
         $encounter = generate_id();
         $data['encounter'] = $encounter;
         $data['uuid'] = UuidRegistry::getRegistryForTable(self::ENCOUNTER_TABLE)->createUuid();
-        $data['date'] = date("Y-m-d");
+        // $data['date'] = date("Y-m-d");
```

Comment:
I recommend changing this to set the date to be the current date IF $data['date'] is empty.  That will make sure we maintain at least some backwards compatibility with API callers who have not been sending the encounter date.

If you add in that fix then the AppointmentService will still work as this insertEncounter function is called there.
https://github.com/openemr/openemr/blob/46c3f55c8dfe860106ca416f1cc3c9089e88e897/src/Services/AppointmentService.php#L388-L405